qiskit/providers/__init__.py

```diff
@@ -110,6 +109,394 @@
+
+    from qiskit.providers import ProviderV1 as Provider
+    from qiskit.providers.providerutils import filter_backends
+
+    from .backend import MyBackend
+
+    class MyProvider(Provider):
+
+        def __init__(self, token=None):
+            super().__init__()
+            self.token = token
+            self.backends = [MyBackend(provider=self)]
+
+        def backends(self, name=None, **kwargs):
+            if name:
+                backends = [
+                    backend for backend in backends if backend.name() == name]
+            return filter_backends(backends, filters=filters, **kwargs)
+
+The key aspects to ensure are present is that any necessary information for
+authentication (if required) are present in the class and that the backends
+method matches the abstract interface and will return initialized Backend
+objects. The rest is up to the specific provider on how to implement.
+
+Backend
+=======
+
+The backend classes are the core to the provider. These class definitions
+define the interface between Qiskit and the hardware or simulator that will
+execute circuits. They also define the interface for a backend to provide the
+necessary information describing the backend to the compiler so that it can
+embed and optimize any circuit to the backend. There are 3 required things
+in every backend object: a backend configuration object set in the
+`__init__()` method, a ``run()`` method, and a ``_default_options()`` method.
+For example, a minimum working example would be something like::
+
+    from qiskit.providers import BackendV1 as Backend
+    from qiskit.providers.models import BackendConfiguration
+    from qiskit.providers import Options
+
+
+    class MVbackend(Backend):
+
+        configuration = {
+            'backend_name': 'my_backend',
+            'backend_version': '1.0.0',
+            'n_qubits': 5,
+            'basis_gates': ['p', 'sx', 'u', 'cx', 'id'],
+            'coupling_map': [[0, 1], [1, 2], [2, 3], [3. 4]],
+            'simulator': False,
+            'local': False,
+            'conditional': True,
+            'open_pulse': False,
+            'memory': True,
+            'max_shots': 8196,
+            'gates': []
+        }
+
+        def __init(self, provider):
+            super().__init__(
+                configuration=BackendConfiguration.from_dict(self.configuration),
+                provider=provider)
+
+        @classmethod
+        def _default_options(cls):
+            return Options(shots=1024, memory=False)
+
+        def run(circuits, **kwargs):
+            # serialize circuits submit to backend and create a job
+            for kwarg in kwargs:
+                if not hasattr(kwarg, self.options):
+                    warnings.warn(
+                        "Option %s is not used by this backend" % kwarg,
+                        UserWarning, stacklevel=2)
+            options = {
+                'shots': kwargs.get('shots', self.options.shots)
```

Comment:
Typo missing `,` at the end of `'shots': kwargs.get('shots', self.options.shots)`